flumine/markets/market.py

```diff
@@ -62,6 +63,49 @@ def close_market(self) -> None:
             extra=self.info,
         )
 
+    def matches_streaming_market_filter(self, market_filter: dict) -> bool:
```

Comment:
This is really ugly

Reply:
Feel free to rename it, I don't care.

Reply:
Not the name, the logic, I would rather it was more comprehensive.

```python
for stream in strategy.streams:
    if market_id in stream._listener.stream._caches:
        return True
return False
```

Obviously the use of hidden methods is ugly but the logic is solid.

Reply:
I wish you posted that code snippet before I went through all the effort of testing against the market filter 😅.

I've replaced the check with your snippet, with added exception handling for `AttributeError` since `stream._listener.stream` can be `None` in some cases. Because the check is so hacky and relies on bflw internals to work, I've also added an integration test covering market catalogue processing.